ksqldb-functional-tests/src/test/resources/historical_plans/elements_-_validate_schema_id_without_elements_OK_-_JSON_SR_SCHEMA/7.1.0_1636006774347/spec.json

```diff
@@ -44,7 +44,9 @@
         "type" : "object",
         "properties" : {
           "c1" : {
-            "type" : "integer"
+            "type" : "integer",
+            "connect.index": 0,
+            "connect.type": "int64"
```

Comment:
Is this compatible with schema without connect meta. A good test might be:
1. Create stream with json_sr format in SR without this fix.
2. Insert into the stream.
3. With this fix, insert into stream again. Since insertion will try to register schema with connect metadata now, we can see if it's compatible with old schema.

I'm ok if you verify this manually.

Reply:
I figured out why QTT is behaving different. Before this fix, QTT was using the plain JSON serialiazer which does not handle SR schema. It just uses the one from the historical plan and serializes the record. 

With the fix, I had to change QTT to use the JsonSchemaConvert (which is used in Prod). In Prod, when the schema ID is present, then it does not register the schema again, but in QTT we register it 'cause the `ConnectSerdeSupplier` does not add the schema ID to the converter. That's what the commit I reverted used to do (add the schema ID to prevent register the schema).

So, I either re-revert the commit that used to do that so the historical plans are unchanged or we stick with these changes. What do you think?

Reply:
We can stick to current code then. Adding logic in test framework in order to make test work seems not right...e ven test framework logic isn't what prod code is doing. cc @suhas-satish , @colinhicks